src/dxtbx/format/FormatROD.py

```diff
@@ -0,0 +1,456 @@
+            ]
+        ).reshape(3, 3)
+        rot_e2 = np.array(
+            [
+                1,
+                0,
+                0,
+                0,
+                np.cos(detector_rotns_rad[1]),
+                np.sin(detector_rotns_rad[1]),
+                0,
+                -np.sin(detector_rotns_rad[1]),
+                np.cos(detector_rotns_rad[1]),
+            ]
+        ).reshape(3, 3)
+        rot_theta = np.array(
+            [
+                np.cos(gonio_angles[1]),
+                0,
+                np.sin(gonio_angles[1]),
+                0,
+                1,
+                0,
+                -np.sin(gonio_angles[1]),
+                0,
+                np.cos(gonio_angles[1]),
+            ]
+        ).reshape(3, 3)
+        detector_axes = rot_theta.dot(rot_e2.dot(rot_e1))
+        # The third axis points to the opposite of XDS.
+        detector_axes[2, :] *= -1
+
+        pixel_size_x = self._bin_header["real_px_size_x"]
+        pixel_size_y = self._bin_header["real_px_size_y"]
+        origin_at_zero = np.array(
+            [
+                -self._bin_header["origin_px_x"] * pixel_size_x,
+                +self._bin_header["origin_px_y"] * pixel_size_y,
+                +self._bin_header["distance_mm"],
+            ]
+        )
+
+        # FIXME: this formula seem to give the right answer but I don't know why.
+        # XDS exporter in IPR's CrysAlisPro seems broken. It writes the same ORGX/Y
+        # regardless of the theta angles.
+        origin = detector_axes.dot(origin_at_zero)
+
+        detector = self._detector_factory.make_detector(
+            "PAD",
+            detector_axes[:, 0],
+            -detector_axes[:, 1],
+            origin,
+            (pixel_size_x, pixel_size_y),
+            (self._txt_header["NX"], self._txt_header["NY"]),
+            (0, self._bin_header["overflow_threshold"]),
+        )  # not sure about min
+
+        return detector
+
+    def get_scan(self, index=None):
+        return Format.get_scan(self)
+
+    def _scan(self):
+        """Return the scan information for this image."""
+
+        for axis in [0, 3]:  # 0 - OMEGA, 3 - PHI: FIXME: is it always PHI scan?
+            start_angle = (
+                self._bin_header["start_angles_steps"][axis]
+                * self._bin_header["step_to_rad"][axis]
+                / np.pi
+                * 180
+            )
+            end_angle = (
+                self._bin_header["end_angles_steps"][axis]
+                * self._bin_header["step_to_rad"][axis]
+                / np.pi
+                * 180
+            )
+            if start_angle != end_angle:
+                break
+
+        return self._scan_factory.single_file(
+            filename=self._image_file,
+            exposure_times=self._bin_header["exposure_time_sec"],
+            osc_start=start_angle,
+            osc_width=end_angle - start_angle,
+            epoch=None,
+        )
+
+    def get_raw_data(self):
```

Comment:
By way of a narrative; spot finding _feels slow_ which I suspect is down to the fact that the `get_raw_data()` method is itself slower than the equivalent e.g. from `CBF` or `HDF5` data.

I mark this as an opportunity for future improvement perhaps as a C++ implementation rather than a bug here.